core/variables_dynamic.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Utility functions for handling variables dynamic.
+ *
+ * @author duzc2dtw@gmail.com (Du Tian Wei)
+ */
+'use strict';
+
+goog.provide('Blockly.VariablesDynamic');
+
+goog.require('Blockly.Variables');
+goog.require('Blockly.Blocks');
+goog.require('Blockly.constants');
+goog.require('Blockly.VariableModel');
+goog.require('Blockly.Workspace');
+goog.require('goog.string');
+
+
+/**
+ * Prompt the user for a new variable name.
+ * @param {string} promptText The string of the prompt.
+ * @param {string} defaultText The default value to show in the prompt's field.
+ * @param {function(?string)} callback A callback. It will return the new
+ *     variable name, or null if the user picked something illegal.
+ */
+Blockly.VariablesDynamic.promptType = function(promptText, defaultText, callback) {
```

Comment:
For now instead of prompting for the user to enter a type can you just use the existing create variable button without a type? We're still working through the UX for how a user picks a type but we probably don't want them to type in a string. This does mean for testing the typed variables need to be created in code.

Reply:
I think it's very important how to select the type.
but before we decide how to make the UX , I think it's not good idea to change the behaviour of  existing button.
the old button should keep Untyped or typed '' , that keep the old behaviour , even after we maked new typed variables . 
that's why we need a new category , and new blocks.

Reply:
I'm not suggesting you change the existing popup. Instead, please add a button to the playground that creates a standard set of variables of different types for testing and just use the existing untyped popup in the toolbox for now.

We don't want to release a UI for creating typed variables that we don't intend to support since that would cause issues for developers if we replace or remove it later.

Reply:
excuse me . 
do you mean that put some buttons in category flyout , with different types ?
e.g. 'Create String Variable','Create Number Variable','Create Color Variable' ?